### PR TITLE
[20.10 backport] Prepare for Go 1.19 upgrade

### PR DESCRIPTION
**Pull request by corhere**
**- What I did**

Prerequisite work to unblock upgrading to Go 1.19

**- How I did it**

Backported:
- https://github.com/docker/cli/pull/3443
  - skipping https://github.com/docker/cli/pull/3443/commits/8c3ae386cd263c5f4fb4c8f7519febedaf1e3281 as golangci-lint v1.49.0 has a linter which flags uses of any deprecated code without needing any configuration
- https://github.com/docker/cli/pull/3708
- https://github.com/docker/cli/pull/3768

**- How to verify it**

- Rebase https://github.com/docker/cli/pull/4032 on top of this branch
- `make -f docker.Makefile lint test-unit` passes

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: cli/command/config/inspect.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func RunConfigInspect(dockerCli command.Cli, opts InspectOptions) error {
 	// check if the user is trying to apply a template to the pretty format, which
 	// is not supported
 	if strings.HasPrefix(f, "pretty") && f != "pretty" {
-		return fmt.Errorf("Cannot supply extra formatting options to the pretty template")
+		return errors.New("cannot supply extra formatting options to the pretty template")
```
Comment (thaJeztah) [inline cli/command/config/inspect.go:55]: Slightly concerned about this commit, as it changes the output of errors. For sure, people "shouldn't" string match these, but it's _possible_, and may be unexpected in a patch release.

**Review by thaJeztah — COMMENTED** [review]
overall looks good; some considerations;

- perhaps we should consider skipping the commit that changes the error messages 
- for the io/ioutil changes; I'd also be fine to add a exclusion rule (I think GolangCi-lint has one for this); while it's marked deprecated, (AFAIK), it won't be removed until Go 2, so in reality, it's only to satisfy the linter, so if we prefer to keep a smaller diff, it's worth considering.

File: cli/command/config/inspect.go (hunk 1 above)
Comment (corhere) [reply cli/command/config/inspect.go:55]: That's fair; I'll replace that commit with a golangci-lint exclude rule

**Comment by corhere** [comment]
> * for the io/ioutil changes; I'd also be fine to add a exclusion rule (I think GolangCi-lint has one for this); while it's marked deprecated, (AFAIK), it won't be removed until Go 2, so in reality, it's only to satisfy the linter, so if we prefer to keep a smaller diff, it's worth considering.

I figure it'd be nice to have to reduce the delta between branches for backports, hopefully making it slightly easier on @neersighted and myself to maintain this branch over the coming months. But that's a trivial matter and the work to backport those changes is a sunk cost so I have no strong opinion one way or the other. I'll leave it up to you and the other reviewers to decide.

**Review by neersighted — APPROVED** [review]

**Comment by thaJeztah** [comment]
Thanks! Yes, I'm fine reducing the diff; the `ioutil` changes are well-understood, and should have no risk in itself. Let me get this one in 👍

**Pull request merged**